lib/java-extras/src/main/java/org/triplea/java/collections/CollectionUtils.java

```diff
@@ -161,6 +162,15 @@ public static <T> T getAny(final Iterable<T> elements) {
     return elements.iterator().next();
   }
 
+  /** Like getAny, but skip elements if possible */
+  public static <T> T getAt(final Iterable<T> elements, final Integer count) {
```

Comment:
I don't quite fully understand the context behind why `count` is significant.
With that caveat, the interface strikes me as odd. Typically the ordering of an iterable is not deterministic, yet we are getting the "n'th" element seemingly? 


Ignoring the 'no next element' issue, this example highlights the difference:

```
// deterministic
getAt(new LinkedHashMap().keySet(), 0)

vs

// not deterministic
getAt(new HashMap().keySet(), 0)
```

At a surface level, this can be somewhat fixed by moving this method to the class that uses it,and making it private and more context specific.

Reply:
> I don't quite fully understand the context behind why `count` is significant.

count represents the number of times `for (int i = 1; i <= numSupportAvailableToApply; i++)` has proceeded through the loop adjusted for the size of `intMap.keySet()`, which can be reduced in getNextAvailableSupporter.

The collection intMap is not being recreated, to my understanding. The same collection is being passed around. 

> At a surface level, this can be somewhat fixed by moving this method to the class that uses it,and making it private and more context specific.

Ok.

Cheers...

Reply:
> count represents the number of times for (int i = 1; i <= numSupportAvailableToApply; i++) has proceeded through the loop adjusted for the size of intMap.keySet(), which can be reduced in getNextAvailableSupporter.

What does it mean for the value to be adjusted by the size of that intmap? It's not clear to me yet what 'count' represents, what is being counted. Is that the count of supporters that have been 'used' up so far?